ironfish/src/account/accounts.ts

```diff
@@ -616,12 +616,18 @@ export class Accounts {
   ): Promise<ReadonlyArray<{ hash: string; note: Note; index: number | null }>> {
     const unspentNotes = []
 
+    const notePromises = []
     for (const transactionMapValue of this.transactionMap.values()) {
-      const result = await this.workerPool.getUnspentNotes(
-        transactionMapValue.transaction.serialize(),
-        [account.incomingViewKey],
+      notePromises.push(
+        this.workerPool.getUnspentNotes(transactionMapValue.transaction.serialize(), [
+          account.incomingViewKey,
+        ]),
```

Comment:
Does the order in which these promises are resolved matter?

Reply:
It shouldn't - all this is really doing is passing the transaction off to the worker threads to try decrypting the notes to find any that you own (can spend). It's being kicked off to the worker threads to keep the main event loop from being blocked by the decrypt code which takes a decent amount of time.

Since a note can only be spent once, order should be unimportant because we then immediately check if the decrypted notes have been spent from our note-nullifier map: https://github.com/iron-fish/ironfish/pull/1518/files#diff-975265306b3138d4067a44c1f74a98ef80249217a05ac91dff96236b482c429bR632

The actual decrypt note function: https://github.com/iron-fish/ironfish/blob/master/ironfish-rust-nodejs/src/structs/note_encrypted.rs#L80-L94